client/src/components/campaignCreatorComponents/CampaignCreator.tsx

```diff
@@ -36,8 +43,81 @@ const CampaignCreator = ({ campaignSettings, setCampaignSettings }: CampaignCrea
         navigate('/twilight-library/dashboard');
     };
 
+    const addToTimeline = (
+        timeline: Record<number, string[]>,
+        nodeKey: keyof NodePillarLists,
+        typeKey: keyof TimelineOptionList,
+    ) => {
+        if (!selectedCampaign.flexible_nemesis_encounters && typeKey === 'nemesis') {
+            // Handle strict Nemesis timeline
+            const nemesis_tier: string = nodeKey.slice(-1);
+            selectedCampaign[nodeKey].forEach((selection: string) => {
+                [1, 2, 3].forEach((level: number) => {
+                    const fight_str: string = `nn${nemesis_tier}_lvl${level}_fight_year`;
+                    const fight_key: keyof NemesisFightYearLists = fight_str as keyof NemesisFightYearLists;
+                    const fight_year: number | null = selectedCampaign[fight_key];
+                    if (fight_year !== null) {
+                        timeline[fight_year].push(`NE - ${selection} lvl ${level}`);
+                    }
+                });
+            });
+            return;
+        }
+
+        selectedCampaign[nodeKey].forEach((selection: string) => {
```

Comment:
I would just break these two blocks out into separate functions

Reply:
just put that section into an else statement